src/compiler/visitorPublic.ts

```diff
@@ -580,20 +581,20 @@ export function visitCommaListElements(elements: NodeArray<Expression>, visitor:
  * @param visitor The callback used to visit each child.
  * @param context A lexical environment context for the visitor.
  */
-export function visitEachChild<T extends Node>(node: T, visitor: Visitor, context: TransformationContext): T;
+export function visitEachChild<T extends Node>(node: T, visitor: Visitor, context?: TransformationContext): T;

```

Comment:
I would rather these be `| undefined` instead of optional, as it's too easy to neglect to supply the `context` when it is needed. We do the same for `NodeFactory.createTempVariable`, where you have to explicitly pass `undefined` for the `recordTempVariable` callback parameter.

Reply:
Sure, I can do that; do you prefer that to exporting the null transformation context? (Just checking, you couldn't make it to the meeting last Friday)

Reply:
I don't have an issue with making `nullTransformationContext` public, but I think the approach of just passing `undefined` works just as well.